lab_2_keywords_cooccurrence/main.py

```diff
@@ -83,12 +120,25 @@ def calculate_word_degrees(candidate_keyword_phrases: KeyPhrases,
 
     In case of corrupt input arguments, None is returned
     """
-    if not type_check(candidate_keyword_phrases, list) or not type_check(content_words, list):
+    if not candidate_keyword_phrases or not content_words:
+        return None
+    if not isinstance(candidate_keyword_phrases, list) or not isinstance(content_words, list):
         return None
-    word_degrees = {}
-    for token in content_words:
-        word_degrees[token] = sum(len(phrase) for phrase in candidate_keyword_phrases if token in phrase)
-    return word_degrees
+    if not check_list_types(candidate_keyword_phrases, tuple) or not check_list_types(content_words, str):
+        return None
+    for sequence in candidate_keyword_phrases:
+        if not check_list_types(sequence, str):
+            return None
+
+    word_degrees_dict = {}
+    for word in content_words:
```

Comment:
I believe you can integrate this for to the for of counting `word_degrees_dict`

Reply:
done